lumberjack.go

```diff
@@ -356,6 +379,57 @@ func (l *Logger) oldLogFiles() ([]logInfo, error) {
 	return logFiles, nil
 }
 
+// compressLog compresses the log with given filename
+// using Gzip compression
+func compressLog(filename string) error {
+
+	// If a blank filename is passed
+	// return false
+	if filename == "" {
+		return errors.New("Filename cannot be blank")
+	}
+
+	rawfile, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer rawfile.Close()
+
+	// Calculate the buffer size for rawfile
+	info, _ := rawfile.Stat()
+
+	// Create the buffer
+	var size int64 = info.Size()
+	rawbytes := make([]byte, size)
```

Comment:
You can't do that.  What if the file is 500mb?  You'll allocate 500mb of RÅM on this one line.  
You should just open a new file, wrap the file in a gzipwriter and use io.copy to copy from the old file to the new.